packages/components/addon/components/hds/application-state/body.hbs

```diff
@@ -0,0 +1,13 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+<div class="hds-application-state__body" ...attributes>
+  {{#if (has-block)}}
+    {{yield}}
+  {{else}}
+    <p class="hds-application-state__body__text hds-typography-body-300">
```

Comment:
A more BEMy way to write this class would be
```suggestion
    <p class="hds-application-state__body-text hds-typography-body-300">
```

BEM doesn't allow `block__element__subelement` notations, but encourages `block__sub-element`.

Same suggestion for the other class names in this PR using a similar notation